File: C#.Net/JDI/Web/JDI Web/Selenium/Elements/Common/Label.cs
using Epam.JDI.Core.Interfaces.Common;
using JDI_Web.Selenium.Base;
using JDI_Web.Selenium.Elements.Base;
using OpenQA.Selenium;

namespace JDI_Web.Selenium.Elements.Common
{
    public class Label: ClickableText, ILabel
    {
        public Label() : this(null) { }
        public Label(By byLocator = null, IWebElement webElement = null, WebBaseElement element = null)
            : base(byLocator, webElement, element:element) { }
    }
}

File: jdi-uitest-win/src/main/java/com/epam/jdi/uitests/win/winnium/elements/common/Label.java
package com.epam.jdi.uitests.win.winnium.elements.common;

import com.epam.jdi.uitests.core.interfaces.common.ILabel;
import com.epam.jdi.uitests.win.winnium.elements.base.Element;
import com.epam.jdi.uitests.win.winnium.elements.base.managers.AttributeTextManager;
import com.epam.jdi.uitests.win.winnium.elements.base.managers.ClickableManager;

public class Label extends Element implements ILabel {
    private ClickableManager clickableManager = new ClickableManager(this);
    private AttributeTextManager attributeTextManager = new AttributeTextManager(this);

    @Override
    public void click() {
        clickableManager.click();
    }

    @Override
    public String getText() {
        return attributeTextManager.getText();
    }

    @Override
    public String waitText(String text) {
        return attributeTextManager.waitText(text);
    }

    @Override
    public String waitMatchText(String regEx) {
        return attributeTextManager.waitMatchText(regEx);
    }

    @Override
    public String getValue() {
        return attributeTextManager.getValue();
    }
}
